src/dex/swaap-v2/swaap-v2.ts

```diff
@@ -0,0 +1,875 @@
+        isSell ? 1 : 2,
+        options.txOrigin,
+        this.augustusAddress,
+        tolerance,
+        this.getQuoteReqParams(),
+      );
+
+      if (!quote.success) {
+        const message = `${this.dexKey}-${
+          this.network
+        }: Failed to fetch RFQ for ${getPairName(
+          normalizedSrcToken.address,
+          normalizedDestToken.address,
+        )}: ${JSON.stringify(quote)}`;
+        this.logger.warn(message);
+        throw new SwaapV2QuoteError(message);
+      } else if (!quote.calldata) {
+        const message = `${this.dexKey}-${
+          this.network
+        }: Failed to fetch RFQ for ${getPairName(
+          normalizedSrcToken.address,
+          normalizedDestToken.address,
+        )}. Missing quote data`;
+        this.logger.warn(message);
+        throw new SwaapV2QuoteError(message);
+      } else if (!quote.router) {
+        const message = `${this.dexKey}-${
+          this.network
+        }: Failed to fetch RFQ for ${getPairName(
+          normalizedSrcToken.address,
+          normalizedDestToken.address,
+        )}. Missing router address`;
+        this.logger.warn(message);
+        throw new SwaapV2QuoteError(message);
+      }
+
+      const srcAmount = BigInt(optimalSwapExchange.srcAmount);
+      const destAmount = BigInt(optimalSwapExchange.destAmount);
+      const quoteTokenAmount = BigInt(quote.amount);
+      const slippageFactor = options.slippageFactor;
+
+      if(side === SwapSide.SELL) {
+        if (
+          quoteTokenAmount <
+          BigInt(
+            new BigNumber(destAmount.toString())
+              .times(slippageFactor)
+              .toFixed(0),
+          )
+        ) {
+          const message = `${this.dexKey}-${this.network}: too much slippage on quote ${side} quoteTokenAmount ${quoteTokenAmount} / destAmount ${destAmount} < ${slippageFactor}`;
+          this.logger.warn(message);
+          throw new SlippageCheckError(message);
+        }
+      } else {
+        if (
+          quoteTokenAmount >
+          BigInt(slippageFactor.times(srcAmount.toString()).toFixed(0))
+        ) {
+          const message = `${this.dexKey}-${
+            this.network
+          }: too much slippage on quote ${side} baseTokenAmount ${srcAmount} / srcAmount ${srcAmount} > ${slippageFactor.toFixed()}`;
+          this.logger.warn(message);
+          throw new SlippageCheckError(message);
+        }
+      }
+
+      const expiryAsBigInt = BigInt(quote.expiration);
+      const minDeadline = expiryAsBigInt > 0 ? expiryAsBigInt : BI_MAX_UINT256;
+
+      return [
+        {
+          ...optimalSwapExchange,
+          data: {
+            router: quote.router,
+            callData: quote.calldata,
+          },
+        },
+        { deadline: minDeadline },
+      ];
+    } catch (e) {
+      if (
+        e instanceof Error &&
+        e.message.endsWith('Request failed with status code 403')
```

Comment:
We should catch not message, but rather status code. We agreed to receive in Telegram chat both: 429 and 403